src/main/java/io/confluent/connect/hdfs/HdfsSinkTask.java

```diff
@@ -77,13 +79,17 @@ public void start(Map<String, String> props) {
 
   @Override
   public void stop() throws ConnectException {
-    hdfsWriter.close();
+    if (hdfsWriter != null) {
+      hdfsWriter.close();
+    }
   }
 
   @Override
   public void put(Collection<SinkRecord> records) throws ConnectException {
     try {
-      hdfsWriter.write(records);
+      if (hdfsWriter != null) {
```

Comment:
How are we getting here when `hdfsWriter` is null? Shouldn't `start` guarantee that it is not null, or result in an error that causes the connector to stop?

Is this actually a bug in the Kafka Connect runtime where it's invoking this even when `start` fails? Or is this because `ConnectException` in `start` does not throw an exception?

This definitely looks dangerous since we're effectively just dropping data with this code.